src/patroni.py

```diff
@@ -61,9 +71,32 @@ def get_postgresql_state(self) -> str:
         Returns:
             running, restarting or stopping.
         """
-        r = requests.get(f"http://{self._pod_ip}:8008/health")
+        r = requests.get(f"http://{self._endpoint}:8008/health")
         return r.json()["state"]
 
+    @property
+    @retry(stop=stop_after_attempt(3), wait=wait_exponential(multiplier=1, min=2, max=10))
+    def cluster_members(self) -> set:
+        """Get the current cluster members."""
+        # Request info from cluster endpoint (which returns all members of the cluster).
+        r = requests.get(f"http://{self._endpoint}:8008/cluster")
+        return set([member["name"] for member in r.json()["members"]])
+
+    @retry(stop=stop_after_attempt(3), wait=wait_exponential(multiplier=1, min=2, max=10))
+    def is_all_members_ready(self) -> bool:
+        """Check if all members are correctly running Patroni and PostgreSQL."""
+        # Request info from cluster endpoint
+        # (which returns all members of the cluster and their states).
+        r = requests.get(f"http://{self._endpoint}:8008/cluster")
+        return all(member["state"] == "running" for member in r.json()["members"])
+
+    @property
+    @retry(stop=stop_after_attempt(3), wait=wait_exponential(multiplier=1, min=2, max=10))
+    def member_started(self) -> bool:
```

Comment:
see comment for restructure of this function

Reply:
Changed on https://github.com/canonical/postgresql-k8s-operator/pull/9/commits/ee0018899c32c6f9b60debcc5179ef39eb6ac02c.